lib/SystemMonitor.js

```diff
@@ -55,25 +72,61 @@ module.exports = kind.singleton(
 	active: false,
 
 	/**
-	* Starts observing the system for idleness.
-	*
-	* @public
+	* @method
+	* @private
+	*/
+	constructor: kind.inherit( function (sup) {
+		return function () {
+			var c, p, f = 0, d = 1000 / this.fpsThreshold;
+
+			this.cb = this.bindSafely(function() {
+				if (!c) {
+					c = utils.perfNow();
+				} else {
+					p = c;
+					c = utils.perfNow();
+					f = ((c - p) < d) ? f + 1 : 0;
+				}
+
+				if (f == this.frameThreshold && this.idle()) {
+					this.active = false;
+					this.emit('idle');
+					if (this.listeners('idle').length === 0) {
+						var idx = dispatcher.features.indexOf(this.bindSafely(this.checkEvent));
```

Comment:
I don't think this will work as you expect. Each call to `bindSafely` returns a new function so you'll never find a previous one with the result of another call.